moc_openshift.py

```diff
@@ -375,10 +379,15 @@ def replace_moc_quota(self, project_name, new_quota):
                     quota_def[quota]["value"] = (
                         str(quota_def[quota]["value"]) + quota_def[quota]["units"]
                     )
-        # RBB: flesh out
-        # else:
-        #
-        # RBB  need to overwrite the value in the quotadef with the ones from the new_quota
+        else:
```

Comment:
I'm confused about the potential values of `new_quota`. If there is no `QuotaMultiplier`, what does it contain? The answer to this question will help me review other parts of this code.

Reply:
So, here is the first document that I wrote outlining this (https://docs.google.com/document/d/1MA7sWLLm-qkOC5uR1sYrozQ6siM7_YhkpUQbcKr96qU/edit#heading=h.rur3h0r9yjx0)

I am sure you remember this being discussed over last summer, I attempted to document the different viewpoints here (https://docs.google.com/document/d/1MA7sWLLm-qkOC5uR1sYrozQ6siM7_YhkpUQbcKr96qU/edit#heading=h.rur3h0r9yjx0).

So what we have in the "moc" quota definition is dictionary of quotas, each having a mangled name that indicates it's scope and it's name(?), ie: "BestEffort:pods" or ":resourcequotas".  Of course, these need to match the configuration file that lists the possible quotas.  This way we can add or remove specific quotas as well as adding and removing subsets of the quotas listed in the configuration file.

This will be needed when we develop an interface that requires this fine grained control.

Reply:
I don't have access to the first document (https://docs.google.com/document/d/1MA7sWLLm-qkOC5uR1sYrozQ6siM7_YhkpUQbcKr96qU/edit#heading=h.rur3h0r9yjx0). I've been referring to https://docs.google.com/document/d/1JoHY88HR-hcvdFEUAwROxvIJyOddGxjuZdeG_Q-v4lk/edit#, which is the only document I've seen.

Reply:
So I will summarize it here to keep the conversation in one spot:

The "Moc Quota" is a dictionary of mangled quota names with the quota values, as in the following:

     {   "Version": "0.9", 
         "Kind": "MocQuota", 
         "ProjectName": "rbb-test", 
         "Quota": {
              ":resourcequotas": 5,
              "BestEffort:pods": 2
         }
     }

The mangled quota names are of the format [quota scope]:[quota name] and the complete list is found in the quota definition json file that is currently the following:

        :persistentvolumeclaims
        :requests.storage
        :requests.ephemeral-storage
        :limits.ephemeral-storage
        :replicationcontrollers
        :resourcequotas
        :services
        :services.loadbalancers
        :services.nodeports
        :secrets
        :configmaps
        :openshift.io/imagestreams
        BestEffort:pods
        NotBestEffort:pods
        NotBestEffort:requests.memory
        NotBestEffort:limits.memory
        NotBestEffort:requests.cpu
        NotBestEffort:limits.cpu
        Terminating:pods
        Terminating:requests.memory
        Terminating:limits.memory
        Terminating:requests.cpu
        Terminating:limits.cpu
        NotTerminating:pods
        NotTerminating:requests.memory
        NotTerminating:limits.memory
        NotTerminating:requests.cpu
        NotTerminating:limits.cpu

This list can also contain a "QuotaMultiplier" which was implemented in the previous pull request for initial quota support.

If the list has a "QuotaMultiplier" element, then all other elements of the list are ignored.

Reply:
Is this sufficient to resolve this conversation?